tools/roslaunch/src/roslaunch/loader.py

```diff
@@ -70,30 +70,36 @@ def convert_value(value, type_):
         #attempt numeric conversion
         try:
             if '.' in value:
-                return float(value)
+                return float(value.strip())
             else:
-                return int(value)
+                return int(value.strip())
         except ValueError as e:
             pass
         #bool
-        lval = value.lower()
+        lval = value.strip().lower()
         if lval == 'true' or lval == 'false':
             return convert_value(value, 'bool')
         #string
         return value
     elif type_ == 'str' or type_ == 'string':
         return value
     elif type_ == 'int':
-        return int(value)
+        return int(value.strip())
     elif type_ == 'double':
-        return float(value)
+        return float(value.strip())
     elif type_ == 'bool' or type_ == 'boolean':
-        value = value.lower()
+        value = value.strip().lower()
         if value == 'true' or value == '1':
             return True
         elif value == 'false' or value == '0':
             return False
         raise ValueError("%s is not a '%s' type"%(value, type_))
+    elif type_ == 'yaml':
+        # - lazy import
+        global yaml
+        if yaml is None:
+            import yaml
+        return yaml.load(value)
```

Comment:
How is that represented on the parameter server? Does this store a complex type (e.g. a nested dictionary) under a single `name`?